apps/skireport/ski_report.star

```diff
@@ -0,0 +1,288 @@
+    "Whistler Blackcomb": "https://www.whistlerblackcomb.com/",
+}
+
+def get_schema():
+    options = [schema.Option(display = resort, value = resort) for resort in RESORT_URLS.keys()]
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Dropdown(
+                id = "resort",
+                name = "Ski Resort",
+                desc = "The resort you want to show. North American Epic Pass Resorts only right now",
+                default = options[0].value,
+                options = options,
+                icon = "faSolidFaMountain",
+            ),
+        ],
+    )
+
+def trimToJSON(js_command):
+    js_command = js_command.split("= ", 1)[1]
+    return js_command[:len(js_command) - 2]
+
+def Getweather_data(resort):
+    """Pulls weather info from the weather page associated with the given resort
+
+    Args:
+        resort (string): The Resort Name as listed in RESORT_URLS
+
+    Returns:
+        dict: a dict containing the temperature, snowfall and description attributes. description is not currently used. Returns None if their is an error fetching the results.
+    """
+
+    url = RESORT_URLS[resort] + WEATHER_URL_STUB
+    r = http.get(url)
+    response = r.body()
+    temperature = None
+    snowfall = None
+    weather_description = None
+    for line in response.splitlines():
+        if line.startswith("    FR.forecasts = "):
+            temp_data = json.decode(trimToJSON(line))
+            temperature = humanize.ftoa(temp_data[0]["CurrentTempStandard"])
+            weather_description = temp_data[0]["ForecastData"][0]["WeatherIconStatus"]
+        if line.startswith("    FR.snowReportData = "):
+            snowfall = json.decode(trimToJSON(line))["TwentyFourHourSnowfall"]["Inches"]
+    if temperature == None:
+        url = RESORT_URLS[resort] + WEATHER_URL_STUB_ALT
+        r = http.get(url)
+        response = r.body()
+        for line in response.splitlines():
+            if line.startswith("    FR.forecasts = "):
+                temp_data = json.decode(trimToJSON(line))
+                temperature = humanize.ftoa(temp_data[0]["CurrentTempStandard"])
+                weather_description = temp_data[0]["ForecastData"][0]["WeatherIconStatus"]
+            if line.startswith("    FR.snowReportData = "):
+                snowfall = json.decode(trimToJSON(line))["TwentyFourHourSnowfall"]["Inches"]
+    if temperature == None or snowfall == None:
+        return None
+    return dict(temperature = temperature, snowfall = snowfall, description = weather_description)
+
+def getTerrain(resort):
+    """Gets the Trail status from a particular resort by scraping the website associated with the resort
+
+    Args:
+        resort (string): The Resort Name as listed in RESORT_URLS
+
+    Returns:
+        _type_: _description_
+    """
+    url = RESORT_URLS[resort] + TERRAIN_URL_STUB
+
+    # Pull an HTML response of the lift status page
+    r = http.get(url)
```

Comment:
We're going to need to add some caching around the http calls to make sure request volume isn't too great.

Reply:
I've added caching every ten minutes for each call in the latest commit